### ISSUE TITLE
Problem with building with shared libraries

### ISSUE THREAD

**Issue by gambiteer** [gambit/gambit#411]
```
firefly:~/programs/gambit/gambit> make modules
make[1]: Entering directory '/home/lucier/programs/gambit/gambit/lib'
make[2]: Entering directory '/home/lucier/programs/gambit/gambit/lib/_define-library'
>>>> building module _define-library for C
../../gsc/gsc: symbol lookup error: ../../gsc/gsc: undefined symbol: ___K_pkg_2d_config_2d_path
>>>> building module _define-library/define-library-expand for C
../../gsc/gsc: symbol lookup error: ../../gsc/gsc: undefined symbol: ___K_pkg_2d_config_2d_path
>>>> building module _define-library/debug for C
../../gsc/gsc: symbol lookup error: ../../gsc/gsc: undefined symbol: ___K_pkg_2d_config_2d_path
../module-common.mk:109: recipe for target 'modules-pre' failed
make[2]: *** [modules-pre] Error 127
make[2]: Leaving directory '/home/lucier/programs/gambit/gambit/lib/_define-library'
makefile:556: recipe for target 'modules-recursive' failed
make[1]: *** [modules-recursive] Error 1
make[1]: Leaving directory '/home/lucier/programs/gambit/gambit/lib'
makefile:317: recipe for target 'modules' failed
make: *** [modules] Error 2
```
Previous commands were
```
    88	16:00	mv gambit gambit-4.9.3-works
    89	16:03	git clone https://github.com/gambit/gambit.git
    90	16:03	cd gambit
    93	16:06	./configure 'CC=gcc -march=native -D___CAN_IMPORT_CLIB_DYNAMICALLY' '--enable-single-host' '--enable-shared' '--enable-multiple-versions'
    94	16:06	make -j 4
```

**Comment by feeley** [comment]
Can you check which one of the configure options is causing the problem?  I have never seen this kind of bug, even during development...

**Comment by gambiteer** [comment]
This fails in the same way:
```
   116	16:50	git clone https://github.com/gambit/gambit.git
   117	16:50	cd gambit
   118	16:51	./configure '--enable-single-host' '--enable-shared' '--enable-multiple-versions'
   119	16:51	make -j4
   120	16:56	make modules
```

**Comment by gambiteer** [comment]
Plain `./configure` works.

**Comment by gambiteer** [comment]
It's
```
   141	17:11	/bin/rm -rf gambit
   142	17:11	git clone https://github.com/gambit/gambit.git
   143	17:12	cd gambit
   144	17:12	./configure --enable-shared
   145	17:12	make -j4
   146	17:16	make modules
```
Evidently, some shared libraries are not being dynamically loaded.

**Comment by lassik** [comment]
I get the following when building master on FreeBSD (using 4.9.3 to help with the bootstrap):

```
_io.c:43544:15: error: use of undeclared label '___L1__23__23_read_2d_bytevector_21_'
___DEF_P_HLBL(___L1__23__23_read_2d_bytevector_21_)
              ^
_io.c:43566:12: error: use of undeclared label '___L7__23__23_read_2d_bytevector_21_'
   ___GOTO(___L7__23__23_read_2d_bytevector_21_)
           ^
```

The `read-bytevector!` implementation was patched recently so maybe something wasn't regenerated after that?

**Comment by gambiteer** [comment]
I changed the title of this issue; do you think that the problem with ##read-bytevector! is caused by shared libraries?  If not, perhaps it should be a separate issue.

**Comment by feeley** [comment]
Please try building from a fresh clone to exclude other sources of build problems.

**Comment by feeley** [comment]
@gambiteer There may be a missing `LD_LIBRARY_PATH=...` prefix when calling `gsc` to compile the modules.

**Comment by gambiteer** [comment]
Looks like it, but I don't have time to search for it now.

**Comment by gambiteer** [comment]
It seems you're right, I did this:
```
make SHELL='sh -x' modules
```
in `lib/_define-library` and I see
```
+ ../../gsc/gsc -:~~bin=../../bin,~~lib=../../lib,~~include=../../include -f -e (let ((x (##build-module "_define-library" (quote C) (quote ((module-ref _define-library)))))) (if (not x) (begin (println "*** module failed to build") (exit 1))))
../../gsc/gsc: symbol lookup error: ../../gsc/gsc: undefined symbol: ___K_pkg_2d_config_2d_path
```
So there should be a LD_LIBRARY_PATH like
```
makefile:	LD_LIBRARY_PATH=$(rootfromhere):${LD_LIBRARY_PATH} $(rootfromhere)/gsc-boot -:~~bin=$(srcdirpfx)$(rootfromhere)/bin,~~lib=$(srcdirpfx)$(rootfromhere)/lib,~~include=$(srcdirpfx)$(rootfromhere)/include -f -warnings -link -flat -o $@ -preload $(MODULES_C)
```
in `lib/makefile`, but I don't know where to add it, sorry.  I'm not even sure where `pkg-config-path` is defined, I see:
```
heine:~/programs/gambit/gambit> grep -R pkg-config-path `find . -name '*.scm'`
./boot/gsc/_front.scm:                         pkg-config pkg-config-path asm)
./boot/gsc/_gsclib.scm:         (pkg-config-path (macro-absent-obj))
./boot/gsc/_gsclib.scm:              (if (##eq? pkg-config-path (macro-absent-obj))
./boot/gsc/_gsclib.scm:                  (macro-force-vars (pkg-config-path)
./boot/gsc/_gsclib.scm:                    pkg-config-path)))
./boot/gsc/_gsclib.scm:               (error "string or string list expected for pkg-config-path: parameter")) ;;;;;;;;;;
./boot/gsc/_gsclib.scm:         pkg-config-path)
./boot/gsc/_gsclib.scm:                                   (##string-or-string-list-join pkg-config-path "\n"))
./boot/gsc/_gsclib.scm:         (pkg-config-path
./boot/gsc/_gsclib.scm:                ((##assq 'pkg-config-path config)
./boot/gsc/_gsclib.scm:            pkg-config-path)
./boot/gsc/_gsclib.scm:         pkg-config-path
./boot/gsc/_gsclib.scm:                           (##string-or-string-list-join pkg-config-path "\n"))
./lib/_define-library/define-library-expand.scm:                    pkg-config-path)
./lib/_eval.scm:    ((cc-options ld-options ld-options-prelude pkg-config pkg-config-path)
./gsi/main.scm:    -pkg-config-path dir        Add directory to pkg-config search path
./gsi/main.scm:    (combine-option! 'pkg-config-path)
./gsi/main.scm:                         (pkg-config-path
./gsi/main.scm:                          (let ((x (##assq 'pkg-config-path options)))
./gsi/main.scm:                                 pkg-config-path: pkg-config-path)
./gsi/main.scm:                                 pkg-config-path: pkg-config-path))
./gsi/main.scm:                             pkg-config-path
./gsi/main.scm:                           (pkg-config-path string))))
./gsc/_front.scm:                         pkg-config pkg-config-path asm)
./gsc/_gsclib.scm:         (pkg-config-path (macro-absent-obj))
./gsc/_gsclib.scm:              (if (##eq? pkg-config-path (macro-absent-obj))
./gsc/_gsclib.scm:                  (macro-force-vars (pkg-config-path)
./gsc/_gsclib.scm:                    pkg-config-path)))
./gsc/_gsclib.scm:               (error "string or string list expected for pkg-config-path: parameter")) ;;;;;;;;;;
./gsc/_gsclib.scm:         pkg-config-path)
./gsc/_gsclib.scm:                                   (##string-or-string-list-join pkg-config-path "\n"))
./gsc/_gsclib.scm:         (pkg-config-path
./gsc/_gsclib.scm:                ((##assq 'pkg-config-path config)
./gsc/_gsclib.scm:            pkg-config-path)
./gsc/_gsclib.scm:         pkg-config-path
./gsc/_gsclib.scm:                           (##string-or-string-list-join pkg-config-path "\n"))
```

**Comment by gambiteer** [comment]
Fixed with c766519e7e692aaf44c8d4562ea3902b92ac32e2.
